src/components/application_manager/src/hmi_language_handler.cc

```diff
@@ -131,8 +131,10 @@ void HMILanguageHandler::on_event(const event_engine::Event& event) {
       is_tts_language_received_ = true;
       break;
     case hmi_apis::FunctionID::BasicCommunication_OnAppRegistered:
-      CheckApplication(
-          std::make_pair(msg[strings::params][strings::app_id].asUInt(), true));
+      if (msg[strings::params].keyExists(strings::app_id)) {
```

Comment:
@dtrunov 
I would prefer  to check if (false) than LOG_ERROR and return. It will help to decrease code nesting.